device: add DEVICE_DT_GET_OR_NULL utility macro

gmarull opened the pull request:
Add a new utility macro to obtain an optional reference to a device. If
the provided node_id is not enabled, the macro falls back to NULL.

gmarull committed:
device: add DEVICE_DT_GET_OR_NULL utility macro

Add a new utility macro to obtain an optional reference to a device. If
the provided node_id is not enabled, the macro falls back to NULL.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: pinmux: stm32: use DEVICE_DT_GET_OR_NULL

Use existing Devicetree macro to obtain optional references to GPIO
ports.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

erwango reviewed (approved):
+1 on my side if you think that `DEVICE_DT_GET_OR` might be useful for others.

ceolin reviewed (changes requested):
Please add a test for this macro in `tests/lib/devicetree/devices/`

gmarull committed:
tests: devicetree: devices: add test for DEVICE_DT_GET_OR_NULL

Add a unit test to verify the correct behavior of the
DEVICE_DT_GET_OR_NULL macro.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented:
> Please add a test for this macro in `tests/lib/devicetree/devices/`

done
